src/pages/login-callback.tsx

```diff
@@ -21,31 +26,62 @@ enum SignUpState {
   COMPLETED,
 }
 
-const hasProfile = async () => {
-  const { error } = await getProfile(getUserId());
-  return !error;
-};
-
 const LoginCallback = () => {
   const navigate = useNavigate();
 
+  const [user, setUser] = useState<User | null>();
+
+  const [showUserLoading, setShowUserLoading] = useState(false);
   const [showConfirmEmailDialog, setShowConfirmEmailDialog] = useState(false);
   const [showProfileDialog, setShowProfileDialog] = useState(false);
   const [showNotificationDialog, setShowNotificationDialog] = useState(false);
   const [showLocationDialog, setShowLocationDialog] = useState(false);
 
   const [currentState, setCurrentState] = useState<SignUpState>();
 
+  const hasProfile = async () => {
+    const { error } = await getProfile(user?.id);
+    return !error;
+  };
+
   const onStateChange = async () => {
     switch (currentState) {
-      case SignUpState.REQUIRE_PROFILE: {
-        const isVerified = getUserId() != 'UNKNOWN';
-        if (!isVerified) {
+      case SignUpState.USER_NOT_LOADED: {
+        setShowUserLoading(true);
+
+        setTimeout(() => {
           setCurrentState(SignUpState.CONFIRM_EMAIL);
-          setShowConfirmEmailDialog(true);
+        }, 500);
+        break;
+      }
+      case SignUpState.CONFIRM_EMAIL: {
+        const session = supabase.auth.session();
+        if (!session || !session.access_token) {
```

Comment:
That's why I used to redirect to the login page after the email confirmation because the session is not set with the `signup()` method if the account is not confirmed yet I think.

Reply:
Mhm then the email confirmation should go to another page and not the login callback.

Reply:
Yes we can add a new page with just the ConfirmEmail Dialog then redirect to `/login`.